src/receipt.rs

```diff
@@ -863,7 +871,7 @@ impl bytes::Deserializable for Receipt {
     fn from_bytes(bytes: &[u8]) -> io::Result<Self> {
         let mut instance = Self::ret(Default::default(), 0, 0, 0);
 
-        instance.write(bytes)?;
+        instance.write_all(bytes)?;
```

Comment:
We shouldn't [write_all](https://doc.rust-lang.org/std/io/trait.Write.html#method.write_all) because the bytes buffer might be bigger than the required.

I honestly think this clipper lint is misleading. We might as well use variable size buffer and eventually use the remainder buffer for other data. If we do so, we will have [ErrorKind::Interrupted](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.Interrupted) when using `write_all` while `write` will just get the bytes required by this instance.

I personally don't like clipper a lot because of many small nits like this one.

Reply:
I see, will be reverted.